src/nvim/ascii.h

```diff
@@ -87,4 +91,48 @@
 # define PATHSEPSTR     "/"
 #endif
 
+static inline bool ascii_iswhite(int) REAL_FATTR_ALWAYS_INLINE REAL_FATTR_CONST;
+static inline bool ascii_isdigit(int) REAL_FATTR_ALWAYS_INLINE REAL_FATTR_CONST;
+static inline bool ascii_isxdigit(int) REAL_FATTR_ALWAYS_INLINE REAL_FATTR_CONST;
+static inline bool ascii_isspace(int) REAL_FATTR_ALWAYS_INLINE REAL_FATTR_CONST;
+
+/// Checks if `c` is a space or tab character.
+static inline bool ascii_iswhite(int c)
+{
+  return c == ' ' || c == '\t';
+}
+
+/// Check whether character is a decimal digit.
+///
+/// Library isdigit() function is officially locale-dependent and, for
+/// example, returns true for superscript 1 (¹) in locales where encoding
+/// contains it in lower 8 bits. Also avoids crashes in case c is below
+/// 0 or above 255: library functions are officially defined as accepting
+/// only EOF and unsigned char values (otherwise it is undefined behaviour)
+/// what may be used for some optimizations (e.g. simple `return
+/// isdigit_table[c];`).
```

Comment:
Now that this very nice description is here, perhaps the other function descriptions could be stripped down and have a `@see {ascii_isdigit}` field added. For example, `ascii_isxdigit`'s comment could be something like this (partially taken from Glibc:
```
/// Checks if `c` is a hexadecimal digit, that is, one of
///           0 1 2 3 4 5 6 7 8 9 a b c d e f A B C D E F.
/// @see {ascii_isdigit}
```